### ISSUE TITLE
Environment.IsPrivilegedProcess unexpectedly returns true on Windows

### ISSUE THREAD

**Issue by iSazonov** [dotnet/runtime#84137]
### Description

Environment.IsPrivilegedProcess return True if process is running with `runas /trustlevel:0x20000` under local administrator account on Windows.

The main reason is that the local administrator account is excluded from UAC.
It is unclear whether this is a bug in the [GetTokenInformation](https://learn.microsoft.com/en-us/windows/win32/api/securitybaseapi/nf-securitybaseapi-gettokeninformation).
If not, then at the very least Environment.IsPrivilegedProcess behavior should be explicitly documented.

Also, I don't know if there is this problem for Service Accounts. In this case, the problem is more unpleasant and must be corrected.

### Reproduction Steps

1. Run process under local administrator account
2. Run subprocess with `runas /trustlevel:0x20000`
3. Call Environment.IsPrivilegedProcess in the subprocess

### Expected behavior

Environment.IsPrivilegedProcess return FALSE

### Actual behavior

Environment.IsPrivilegedProcess return TRUE

### Configuration

.Net 8.0 Preview 2

**Comment by tannergooding** [comment]
Docs issue as we just defer to the underlying OS.

**Comment by A9G-Data-Droid** [comment]
What is the correct method for testing for elevation?

`Environment.IsPrivilegedProcess`

or

`if (principal.IsInRole(WindowsBuiltInRole.Administrator))`

What is the method that Task Manager uses for it's elevated property?

**Comment by BinToss** [comment]
> What is the method that Task Manager uses for it's elevated property?

Honestly, I'm unsure.

Taskmgr.exe statically imports the functions called by `System.Environment.IsPrivilegedProcess`
- `GetTokenInformation` from `api-ms-win-security-base-l1-1-0.dll (kernelbase.dll)` 
- `NtQueryTokenInformation` from `ntdll.dll`
- `LookupPrivilegeValueW` from `api-ms-win-security-lsalookup-l2-1-0.dll (advapi32.dll)`

But it also statically imports `CheckTokenMembership`—the function called by `IsInRole`.

I'd wager it uses the former rather than the latter for checking process elevation. See the next two sections for details.

---

> `Environment.IsPrivilegedProcess`

tl;dr This appears to be the most reliable solution.

On Windows, this getter's call-chain involves the native function `GetTokenInformation` with the `TOKEN_INFORMATION_CLASS.TokenElevation` flag.

<details><summary>.NET Runtime source code</summary>

https://github.com/dotnet/runtime/blob/fd8933aac237d2f3103de071ec4bc1547bfef16c/src/libraries/System.Private.CoreLib/src/System/Environment.cs#L50-L64
https://github.com/dotnet/runtime/blob/fd8933aac237d2f3103de071ec4bc1547bfef16c/src/libraries/System.Private.CoreLib/src/System/Environment.Windows.cs#L89-L115
https://github.com/dotnet/runtime/blob/fd8933aac237d2f3103de071ec4bc1547bfef16c/src/libraries/Common/src/Interop/Windows/Advapi32/Interop.TOKEN_ELEVATION.cs#L4-L14
</details>

According to [System Informer](https://github.com/winsiderss/systeminformer)...
- `true` can have the type "Default" or "Full"
- `false` can have the type "Default" or "Limited"

---

> `if (principal.IsInRole(WindowsBuiltInRole.Administrator))`

tl;dr `IsInRole` is unreliable for determining process elevation.

Although this method is used by [UACHelper's `IsElevated` method](https://github.com/falahati/UACHelper/blob/master/UACHelper/UACHelper.cs#L82-L89), System Informer reveals that a process can be elevated _without_ the `BUILTIN\Administrators` role!
![Image](https://github.com/user-attachments/assets/5d2736a0-1cf8-49aa-bf87-ae0d47caf5f5)

So, `IsInRole` is only correct when a process is run by a local user. It is incorrect regarding processes run by `NT AUTHORITY\LOCAL SERVICE`, `NT AUTHORITY\SYSTEM`, et cetera. The only exception I can find is the Registry process. It's non-elevated and has the `BUILTIN\Administrators` role, but with "deny all" for permissions.

Also, according [this comment on StackOverflow](https://stackoverflow.com/questions/1220213/detect-if-running-as-administrator-with-or-without-elevated-privileges#comment19210300_4497572), `IsInRole(WindowsBuiltInRole.Administrator)` returns `false` for processes elevated by "Domain Administrator" credentials. The method's behavior is correct, but undesired for checking process elevation.

Regarding `IsInRole`'s call chain, it relies on the native function [`CheckTokenMembership`](https://learn.microsoft.com/en-us/windows/win32/api/securitybaseapi/nf-securitybaseapi-checktokenmembership) from "Advapi32.dll".

<details><summary>.NET Runtime source code</summary>

https://github.com/dotnet/runtime/blob/fd8933aac237d2f3103de071ec4bc1547bfef16c/src/libraries/System.Security.Principal.Windows/src/System/Security/Principal/WindowsPrincipal.cs#L114-L180
https://github.com/dotnet/runtime/blob/fd8933aac237d2f3103de071ec4bc1547bfef16c/src/libraries/Common/src/Interop/Windows/Advapi32/Interop.CheckTokenMembership.cs#L4-L19
</details>
